api/comments/permissions.py

```diff
@@ -29,6 +31,8 @@ def has_object_permission(self, request, view, obj):
         auth = get_user_auth(request)
         comment = obj
         node = obj.node
+        if has_anonymous_link(node, auth):
```

Comment:
I talked with Nici, and this is probably not the best way of dealing with this problem, because it'll be a breaking change to the API. Instead, if the view is hit with an anonymous view only link, have it go through the `content` field and look for all the things that look like `[@xxxxxx](xxxxxx)` with stuff instead of just xxxxxx and replace them with something like _Anonymous User Mention_. Ensure that this isn't being hit by the OSF in the case of an anonymous view only link (so make sure the calls to v2/nodes/nid/comments/ is in the `comment_pane_template.mako` and not anywhere above that, which I'm sure it is but just to be sure).

Alternately, you could play around with making a custom text field that can check for if it's anonymous and run a regular expression on the field. That would be better but I don't know how difficult it would be. But it's worth exploring at least some.